### PR TITLE
Partial fix for #607 LiblineaR

### PR DESCRIPTION
**Pull request by florianfendt**
@berndbischl 
This should fix the bug you reported in Issue #607 concerning the LiblineaR Learners

### CODE REVIEW

**Comment by berndbischl** [comment]
Hi Flo, can you summarize where the actual error was that was reported?
Was it simply because only want a certain "type" for the subgroup and this was not set in par.vals? so always type = 0 was chosen?

otherwise the code looks ok

**Comment by florianfendt** [comment]
You nailed it. I didn't recognize that earlier because "type" was always set in the unit tests.
Adjusted the tests accordingly.

**Comment by berndbischl** [comment]
why is travis broken here? I will restart

**Comment by berndbischl** [comment]
Shouldnt we use the FunctionLearnerParam? 
Untyped is also OK, but feval is a function right?

Also we have to add to NEWS and squash commits to one

But we can do this too if you dont want

**Comment by berndbischl** [comment]
The errors seem to be related to ggplot2? I guess this is the update we see now?



```

1. Failure (at test_base_generateCalibration.R#16): generateCalibrationData ----
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to length(unique(cd$data$Learner))
0 - 1 == -1

2. Failure (at test_base_generateCalibration.R#30): generateCalibrationData ----
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to length(unique(cd$data$Learner))
0 - 1 == -1

3. Failure (at test_base_generateCalibration.R#40): generateCalibrationData ----
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to length(unique(cd$data$Learner))
0 - 2 == -2

4. Failure (at test_base_generateCalibration.R#50): generateCalibrationData ----
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to length(unique(cd$data$Learner))
0 - 2 == -2

5. Failure (at test_base_generateLearningCurve.R#13): generateLearningCurve ----
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to length(r$measures)
0 - 2 == -2

6. Failure (at test_base_generateLearningCurve.R#26): generateLearningCurve ----
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to length(r$measures)
0 - 2 == -2

7. Failure (at test_base_generateLearningCurve.R#38): generateLearningCurve ----
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to length(r$measures)
0 - 2 == -2

8. Failure (at test_base_generatePartialPrediction.R#23): generatePartialPredictionData 
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to nfacet
0 - 2 == -2

9. Failure (at test_base_generatePartialPrediction.R#42): generatePartialPredictionData 
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to nfacet
0 - 2 == -2

10. Failure (at test_base_generatePartialPrediction.R#58): generatePartialPredictionData 
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to nfeat
0 - 2 == -2

11. Failure (at test_base_generatePartialPrediction.R#78): generatePartialPredictionData 
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to nfacet
0 - 3 == -3

12. Failure (at test_base_generatePartialPrediction.R#99): generatePartialPredictionData 
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to nfeat
0 - 2 == -2

13. Failure (at test_base_generatePartialPrediction.R#113): generatePartialPredictionData 
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to nfacet
0 - 2 == -2

14. Failure (at test_base_generatePartialPrediction.R#133): generatePartialPredictionData 
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to nfeat
0 - 2 == -2

15. Failure (at test_base_generateThreshVsPerf.R#14): generateThreshVsPerfData -
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to length(pvs$measures)
0 - 2 == -2

16. Failure (at test_base_generateThreshVsPerf.R#30): generateThreshVsPerfData -
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to length(pvs$measures)
0 - 2 == -2

17. Failure (at test_base_generateThreshVsPerf.R#53): generateThreshVsPerfData -
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to length(pvs$measures)
0 - 2 == -2

18. Failure (at test_base_generateThreshVsPerf.R#77): generateThreshVsPerfData -
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to length(pvs$measures)
0 - 2 == -2

19. Failure (at test_base_generateThreshVsPerf.R#85): generateThreshVsPerfData -
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to length(unique(pvs$data$learner))
0 - 2 == -2

20. Failure (at test_base_generateThreshVsPerf.R#92): generateThreshVsPerfData -
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to length(unique(pvs$data$learner))
0 - 2 == -2

21. Failure (at test_base_generateThreshVsPerf.R#106): generateThreshVsPerfData 
length(XML::getNodeSet(doc, grey.xpath, "svg")) not equal to length(pvs.custom$measures)
0 - 1 == -1

22. Error: BenchmarkResult -----------------------------------------------------
(converted from warning) the XPath query has no namespace, but the target document has a default namespace. This is often an error and may explain why you obtained no results
1: withCallingHandlers(eval(code, new_test_environment), error = capture_calls, message = function(c) invokeRestart("muffleMessage"))
2: eval(code, new_test_environment)
3: eval(expr, envir, enclos)
4: expect_equal(length(XML::getNodeSet(doc, grey.xpath, "svg")), length(getBMRTaskIds(res))) at test_base_plotBMRBoxplots.R:14
5: expect_that(object, equals(expected, label = expected.label, ...), info = info, label = label)
6: condition(object)
7: compare(actual, expected, ...)
8: XML::getNodeSet(doc, grey.xpath, "svg")
9: xpathApply(doc, path, fun, ..., namespaces = namespaces, sessionEncoding = sessionEncoding, 
       addFinalizer = addFinalizer)
10: xpathApply.XMLInternalDocument(doc, path, fun, ..., namespaces = namespaces, sessionEncoding = sessionEncoding, 
       addFinalizer = addFinalizer)
11: warning("the XPath query has no namespace, but the target document has a default namespace. This is often an error and may explain why you obtained no results")
12: .signalSimpleWarning("the XPath query has no namespace, but the target document has a default namespace. This is often an error and may explain why you obtained no results", 
       quote(xpathApply.XMLInternalDocument(doc, path, fun, ..., namespaces = namespaces, 
           sessionEncoding = sessionEncoding, addFinalizer = addFinalizer)))
13: withRestarts({
       .Internal(.signalCondition(simpleWarning(msg, call), msg, call))
       .Internal(.dfltWarn(msg, call))
   }, muffleWarning = function() NULL)
14: withOneRestart(expr, restarts[[1L]])
15: doWithOneRestart(return(expr), restart)

23. Error: critDifferences -----------------------------------------------------
Unknown parameters: vjust
1: withCallingHandlers(eval(code, new_test_environment), error = capture_calls, message = function(c) invokeRestart("muffleMessage"))
2: eval(code, new_test_environment)
3: eval(expr, envir, enclos)
4: plotCritDifferences(r1) at test_base_plotCritDifferences.R:26
5: annotate("segment", x = cd.x + cd, xend = cd.x - cd, y = cd.y, yend = cd.y, alpha = 0.5, 
       color = "darkgrey", size = 2, vjust = obj$cd.info$textvjust)
6: layer(geom = geom, params = list(na.rm = na.rm, ...), stat = StatIdentity, position = PositionIdentity, 
       data = data, mapping = aes_all(names(data)), inherit.aes = FALSE, show.legend = FALSE)
7: stop("Unknown parameters: ", paste(extra, collapse = ", "), call. = FALSE)

24. Error: plotLearnerPrediction -----------------------------------------------
(converted from warning) `show_guide` has been deprecated. Please use `show.legend` instead.
1: withCallingHandlers(eval(code, new_test_environment), error = capture_calls, message = function(c) invokeRestart("muffleMessage"))
2: eval(code, new_test_environment)
3: eval(expr, envir, enclos)
4: plotLearnerPrediction("classif.rpart", multiclass.task, gridsize = gs) at test_base_plotLearnerPrediction.R:5
5: geom_tile(data = grid, mapping = aes_string(fill = target, alpha = ".prob.pred.class"), 
       show_guide = TRUE)
6: layer(data = data, mapping = mapping, stat = stat, geom = GeomTile, position = position, 
       show.legend = show.legend, inherit.aes = inherit.aes, params = list(na.rm = na.rm, 
           ...))
7: warning("`show_guide` has been deprecated. Please use `show.legend` instead.", call. = FALSE)
8: .signalSimpleWarning("`show_guide` has been deprecated. Please use `show.legend` instead.", 
       quote(NULL))
9: withRestarts({
       .Internal(.signalCondition(simpleWarning(msg, call), msg, call))
       .Internal(.dfltWarn(msg, call))
   }, muffleWarning = function() NULL)
10: withOneRestart(expr, restarts[[1L]])
11: doWithOneRestart(return(expr), restart)

testthat results ================================================================
OK: 1986 SKIPPED: 1 FAILED: 24
1. Failure (at test_base_generateCalibration.R#16): generateCalibrationData 
2. Failure (at test_base_generateCalibration.R#30): generateCalibrationData 
3. Failure (at test_base_generateCalibration.R#40): generateCalibrationData 
4. Failure (at test_base_generateCalibration.R#50): generateCalibrationData 
5. Failure (at test_base_generateLearningCurve.R#13): generateLearningCurve 
6. Failure (at test_base_generateLearningCurve.R#26): generateLearningCurve 
7. Failure (at test_base_generateLearningCurve.R#38): generateLearningCurve 
8. Failure (at test_base_generatePartialPrediction.R#23): generatePartialPredictionData 
9. Failure (at test_base_generatePartialPrediction.R#42): generatePartialPredictionData 
1. Failure (at test_base_generatePartialPrediction.R#58): generatePartialPredictionData 
2. Failure (at test_base_generatePartialPrediction.R#78): generatePartialPredictionData 
3. Failure (at test_base_generatePartialPrediction.R#99): generatePartialPredictionData 
4. Failure (at test_base_generatePartialPrediction.R#113): generatePartialPredictionData 
5. Failure (at test_base_generatePartialPrediction.R#133): generatePartialPredictionData 
6. Failure (at test_base_generateThreshVsPerf.R#14): generateThreshVsPerfData 
7. Failure (at test_base_generateThreshVsPerf.R#30): generateThreshVsPerfData 
8. Failure (at test_base_generateThreshVsPerf.R#53): generateThreshVsPerfData 
9. Failure (at test_base_generateThreshVsPerf.R#77): generateThreshVsPerfData 
1. Failure (at test_base_generateThreshVsPerf.R#85): generateThreshVsPerfData 
2. Failure (at test_base_generateThreshVsPerf.R#92): generateThreshVsPerfData 
3. Failure (at test_base_generateThreshVsPerf.R#106): generateThreshVsPerfData 
4. Error: BenchmarkResult 
5. Error: critDifferences 
6. Error: plotLearnerPrediction 
7. ...


```

**Comment by berndbischl** [comment]
new restart

**Comment by florianfendt** [comment]
@berndbischl 
I put a note to NEWS and rebased.
I think we're done here?

**Comment by berndbischl** [comment]
yes thx.
I really hate posting this things, and you dont need to change now, but pls also watch stuff like the indentation of the ) in unit tests.

**Comment by florianfendt** [comment]
oh yeah you're right should have seen that, sorry.

**Comment by berndbischl** [comment]
it really does not matter that much, but as i am wasting all of our time with the messages here, i will now fix it in master :)

i am just saying this so you keep an eye for it next time. the problem is my eye always "stops" in these cases then an I can less well concentrate on the real code :)

**Comment by berndbischl** [comment]
done fixing indentation
